### PR TITLE
Move FieldParser construction burden to FieldParserFactory

### PR DESCRIPTION
**Pull request by john-n-smith**
@wpillar, back to this! I've created this separate PR to address the main issue we came up against in #19. This was having to call an initialisation method after construction of a `FieldParser`, before it was in a usable state. The initialisation method was `parse`, although no parsing took place.

This solution moves the burden of `FieldParser` construction to `FieldParserFactory`. The resultant Factory/Builder hybrid  means we have a much more robust way of instantiating a `FieldParser`.

This approach has the added benefit of moving the `FieldParserRegistry` logic to the builder too, which makes more sense than the `FieldParser` adding itself to the Registry.

`ParserDocument` still requires the init method `setXml` to be called (formally `parse`), but I don't think the extra complexity of a builder for this is worthwhile.

Old tests have been removed. They've been re-written in #22.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ before_script:
   - travis_retry composer update --no-interaction --prefer-source
 
 script:
-  - vendor/bin/phpcs -p --warning-severity=0 src/ tests/
+  - vendor/bin/phpcs -p --warning-severity=0 src/
```
Comment (wpillar) [inline .travis.yml]: LOL

**Comment by john-n-smith** [comment]
=) It's temporary!

File: .travis.yml (hunk 1 above)
Comment (john-n-smith) [inline .travis.yml]: =) It's temporary!

**Comment by wpillar** [comment]
Tbh I've lost all context of this work so I'm reviewing cold. Overall the changes look fine, but I'm still concerned about some of these setters being required.

For example, this is perfectly valid, but will error:

```php
$fieldParserFixedText = new FieldParserFixedText();
$fieldParserFixedText->getText();
```

Because of these lines:

https://github.com/graze/ciff-renderer/blob/08e245a8ca5aa30a0b72cbe89d2dce9af5b679c4/src/Parser/FieldParser/FieldParserFixedText.php#L39

https://github.com/graze/ciff-renderer/blob/08e245a8ca5aa30a0b72cbe89d2dce9af5b679c4/src/Parser/FieldParser/FieldParserFixedText.php#L43

It's like writing code that will compile but won't run. I shouldn't be able to create such objects that compile but error at runtime.

Do you see where I'm coming from?

**Comment by john-n-smith** [comment]
Superseded by #31.

**Pull request closed without merging**